mipengine/node/monetdb_interface/common_actions.py

```diff
@@ -289,16 +245,16 @@ def _delete_table_by_type_and_context_id(table_type: str, context_id: str):
     context_id : str
         The id of the experiment
     """
-    cursor.execute(
+    table_names_and_types = MonetDB().execute_with_result(
         f"""
         SELECT name, type FROM tables 
         WHERE name LIKE '%{context_id.lower()}%'
         AND tables.type = {str(_convert_mip2monet_table_type(table_type))} 
         AND system = false
         """
     )
-    for table in cursor.fetchall():
+    for table in table_names_and_types:
         if table[1] == 1:
```

Comment:
table is a pair of name, type.
So table[1] is the type of the pair.

Reply:
I didn't expect an answer here, but a change in the code.
Don't use mysterious numbers in the code. Define constants instead or use `enum.Enum` 
```
MY_TYPE = 1
MY_OTHER_TYPE = 2
...

for _, type in cursor.fetchall():
    if type == MY_TYPE:
        do_something()
```

Reply:
@jassak If you didn't expect an answer, don't make a question, not everyone knows what you expect. Suggestions are helpful, rhetorical questions are not.

@KFilippopolitis 
In addition to @jassak 's suggestion of unpacking, I think that the _convert_monet2mip_table_type method can be used to avoid hard coding the type of the table.

Reply:
Thanks for your points, code is a lot more readable this way.